src/scipp/html/formatting_datagroup_html.py

```diff
@@ -0,0 +1,123 @@
+    shape_repr = ''
+    unit = ''
+    dtype_str = ''
+    preview = ''
+    parent_obj_str = ''
+    objtype_str = type(var).__name__
+    if isinstance(var, (Dataset, DataArray, Variable, DataGroup)):
+        parent_obj_str = "scipp"
+        shape_repr = _make_shape_repr(var)
+        if isinstance(var, Dataset):
+            preview = _make_dataset_summary(var)
+        else:
+            if var.unit is not None:
+                unit = '𝟙' if var.unit == dimensionless else str(var.unit)
+            dtype_str = str(var.dtype)
+            preview = str(var.value) if len(var.dims) == 0 else ""
+    else:
+        if isinstance(var, np.ndarray):
+            shape_repr = escape("()")
+            parent_obj_str = "numpy"
+            preview = f"shape={var.shape}, dtype={var.dtype}"
+        if isinstance(var, (str, int, float)):
+            preview = str(var)
+
+    html_tpl = load_atomic_tpl()
+    return Template(html_tpl).substitute(depth=depth,
+                                         name=escape(escape(name)),
+                                         parent=escape(parent_obj_str),
+                                         objtype=escape(objtype_str),
+                                         shape_repr=shape_repr,
+                                         dtype=escape(dtype_str),
+                                         unit=escape(unit),
+                                         preview=preview)
+
+
+def _collapsible_summary(var: DataGroup, name_spaces: list):
+    name = name_spaces[-1]
+    parent_type = "scipp"
+    objtype = type(var).__name__
+    shape_repr = _make_shape_repr(var)
+    checkbox_id = escape("summary-" + str(uuid.uuid4()))
+    depth = len(name_spaces)
+    subsection = _datagroup_detail(var, name_spaces)
+    html_tpl = load_collapsible_tpl()
+
+    return Template(html_tpl).substitute(name=escape(str(name)),
+                                         parent=escape(parent_type),
+                                         objtype=escape(objtype),
+                                         shape_repr=shape_repr,
+                                         summary_section_id=checkbox_id,
+                                         depth=depth,
+                                         checkbox_status='',
+                                         subsection=subsection)
+
+
+def _datagroup_detail(dg: DataGroup, name_spaces: list = None):
+    if name_spaces is None:
+        name_spaces = []
+    summary_rows = []
+    for name, item in dg.items():
+        if isinstance(item, DataGroup):
+            collapsible_row = _collapsible_summary(item, name_spaces + [name])
+            summary_rows.append(collapsible_row)
+        else:
+            summary_rows.append(
+                _summarize_atomic_variable(item, name, depth=len(name_spaces)))
+
+    dg_detail_tpl = Template(load_dg_detail_tpl())
+    return dg_detail_tpl.substitute(summary_rows=''.join(summary_rows))
+
+
+def datagroup_repr(dg: DataGroup):
```

Comment:
```suggestion
def datagroup_repr(dg: DataGroup) -> str:
```
I guess? Also other functions lack annotations for their return type.